src/PKSim.Presentation/Services/RenameBuildingBlockTask.cs

```diff
@@ -241,16 +241,18 @@ private FormulaUsablePath formulaUsablePathReferencing(IFormula formula, IFormul
 
       private IEnumerable<IContainer> getContainersToRename(Simulation simulation, IPKSimBuildingBlock templateBuildingBlock, string oldContainerName)
       {
+         var containersToRename = new List<IContainer>();
+
          var rootContainer = simulation.Model.Root.EntityAt<IContainer>(getContainerPathToRename(templateBuildingBlock.BuildingBlockType));
          if (templateBuildingBlock.BuildingBlockType != PKSimBuildingBlockType.Formulation)
-            yield return rootContainer.Container(oldContainerName);
+            containersToRename.Add(rootContainer.Container(oldContainerName));
          else
          {
-            foreach (var formulationContainer in rootContainer.GetAllChildren<IContainer>(x => x.IsNamed(oldContainerName) && x.ContainerType == ContainerType.Formulation))
-            {
-               yield return formulationContainer;
-            }
+            rootContainer.GetAllChildren<IContainer>(x => x.IsNamed(oldContainerName) && x.ContainerType == ContainerType.Formulation).Each(formulationContainer =>
```

Comment:
you can't have children that are null. (especially because you are checking on their properties already). So return right away here
And if not, use AddRange